homeassistant/components/esphome/config_flow.py

```diff
@@ -111,16 +112,7 @@ def _name(self, value: str) -> None:
         self.context[CONF_NAME] = value
         self.context["title_placeholders"] = {"name": self._name}
 
-    def _set_user_input(self, user_input: dict[str, Any] | None) -> None:
-        if user_input is None:
-            return
-        self._host = user_input[CONF_HOST]
-        self._port = user_input[CONF_PORT]
-
-    async def _async_try_fetch_device_info(
-        self, user_input: dict[str, Any] | None
-    ) -> FlowResult:
-        self._set_user_input(user_input)
```

Comment:
There are 2 places that call this function, and only one passes user input. Moved the setting of `self._host` and `self._port` there.